hack/tools/e2e-profiling/collect-profiles.sh

```diff
@@ -0,0 +1,304 @@
+}
+
+log_success() {
+    echo -e "${GREEN}[SUCCESS]${NC} $*"
+}
+
+log_warn() {
+    echo -e "${YELLOW}[WARN]${NC} $*"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $*"
+}
+
+# Validate profile mode
+case "${PROFILE_MODE}" in
+    both|heap|cpu)
+        ;;
+    *)
+        log_error "Invalid E2E_PROFILE_MODE: ${PROFILE_MODE}"
+        log_error "Valid options: both, heap, cpu"
+        exit 1
+        ;;
+esac
+
+# Create output directory
+mkdir -p "${OUTPUT_DIR}"
+log_info "Output directory: ${OUTPUT_DIR}"
+log_info "Profile mode: ${PROFILE_MODE}"
+
+# PIDs to track
+declare -A PF_PIDS
+declare -A PODS
+
+# Cleanup function
+cleanup() {
+    log_info "Cleaning up..."
+    for component in "${!PF_PIDS[@]}"; do
+        if [ -n "${PF_PIDS[$component]:-}" ]; then
+            log_info "Killing port-forward for ${component} (PID: ${PF_PIDS[$component]})"
+            kill "${PF_PIDS[$component]}" 2>/dev/null || true
+        fi
+    done
+}
+
+trap cleanup EXIT INT TERM
+
+# Wait for namespace to exist first
+log_info "Waiting for namespace ${NAMESPACE} to exist..."
+TIMEOUT=300  # 5 minutes
+ELAPSED=0
+NAMESPACE_CHECK_INTERVAL=5
+while [ $ELAPSED -lt $TIMEOUT ]; do
+    if kubectl get namespace "${NAMESPACE}" >/dev/null 2>&1; then
+        log_success "Namespace ${NAMESPACE} exists"
+        break
+    fi
+    sleep $NAMESPACE_CHECK_INTERVAL
+    ELAPSED=$((ELAPSED + NAMESPACE_CHECK_INTERVAL))
+done
+
+if [ $ELAPSED -ge $TIMEOUT ]; then
+    log_error "Namespace ${NAMESPACE} did not appear within ${TIMEOUT} seconds"
+    exit 1
+fi
+
+# Set up port forwarding for all components
+for component in "${!COMPONENTS[@]}"; do
+    # Parse component configuration
+    IFS=';' read -ra CONFIG <<< "${COMPONENTS[$component]}"
+    DEPLOYMENT=""
+    POD_LABEL=""
+    PPROF_PORT=""
+    LOCAL_PORT=""
+
+    for item in "${CONFIG[@]}"; do
+        key="${item%%=*}"
+        value="${item#*=}"
+        case "$key" in
+            deployment) DEPLOYMENT="$value" ;;
+            label) POD_LABEL="$value" ;;
+            port) PPROF_PORT="$value" ;;
+            local_port) LOCAL_PORT="$value" ;;
+        esac
+    done
+
+    log_info "Setting up ${component}..."
+
+    # Wait for deployment to exist first
+    log_info "Waiting for deployment ${DEPLOYMENT} to be created in namespace ${NAMESPACE}..."
+    TIMEOUT=300
+    ELAPSED=0
+    while ! kubectl get deployment -n "${NAMESPACE}" "${DEPLOYMENT}" &> /dev/null; do
+        if [ $ELAPSED -ge $TIMEOUT ]; then
+            log_error "Deployment ${DEPLOYMENT} was not created within ${TIMEOUT} seconds"
+            exit 1
+        fi
+        sleep 2
+        ELAPSED=$((ELAPSED + 2))
+    done
+    log_success "Deployment ${DEPLOYMENT} exists"
+
+    # Now wait for deployment to become available
+    log_info "Waiting for deployment ${DEPLOYMENT} to become available..."
+    if ! kubectl wait --for=condition=Available -n "${NAMESPACE}" deployment "${DEPLOYMENT}" --timeout=300s; then
```

Comment:
we are going to wait until condition `Available` appears, but we are not checking its status, thus we might continue even if the status is set to `False`. We should actually continue only when it is set to `True`

Reply:
Thanks, my bad.